gitfile/tokenfetcher.go

```diff
@@ -0,0 +1,31 @@
+package gitfile
+
+// HeaderStruct is the simple struct to carry authentication string from different suppliers
+type HeaderStruct struct {
+	Authorization string `json:"Authorization"`
+}
+
+// TokenFetcher is the interface for the authentication token suppliers which are provides tokens as a HeaderStruct
+// instances
+type TokenFetcher interface {
+	BuildHeader(repoUrl string) HeaderStruct
+}
+
+var TokenFetchers []TokenFetcher
+
+func init() {
```

Comment:
The caller has no control over how and where from we load the tokens.. IMHO this should be in the control of the library user, not hardcoded on the library side.

The library could provide some implementations of how the token might be obtained, but the caller should be in control which one(s) is(are) used. Also, the user of the library should be able to provide their own `TokenFetcher` so that we don't prescribe where and how the token should be stored.